### ISSUE TITLE
Improved expandable multi-level sidebar

### ISSUE THREAD

**Issue by hamishwillee** [3drobotics/sphinx_3dr_theme#8]
## Previously raised in wrong repo. I am reopening here because we have hit the limitations of the current sidebar.

Make the sidebar expandable like the one [this site](http://kripken.github.io/emscripten-site/docs/getting_started/index.html). 

What this allows that we don't have now:
- _Full TOC_ is displayed for current page in the sidebar.
- Current page is highlighted in sidebar, providing context
- Deeper nesting - at the moment we really only have 2 levels, and we really need 3 or 4.
- Expansion of the tree to the current point.

In summary, this displays a more TOC detail for the current page, while still giving the overview/context of the page within the hierarchy.  This is a much better sidebar if you have a bigger documentation set, with some levels of nesting.

**Comment by hamishwillee** [comment]
Closed by #17
